### PR TITLE
feat(desktop/wallet) Removed dynamic scoping from wallet

### LINKED ISSUE #4245: [refactor] remove dynamic scoping from wallet module

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #4245

### What does the PR do
Removed all dynamic scoping parts in wallet section plus

- Renamed Config to Global and added more functions there
- Moved changeAppSectionBySectionType function to Global
  and updated all places where is used to call it from
  Global instead
- Moved openLink function to Global and updated all places
  where is used to call it from Global instead
- Moved errorSound to Global, introduced playErrorSound
  function  and updated all places where is used to call
  this function from Global instead

### Affected areas
Browser, Chat, Profile, Wallet, shared

### CODE REVIEW

File: ui/app/AppLayouts/Profile/views/ContactsView.qml [hunk 1]
```diff
@@ -259,12 +259,12 @@ Item {
             onCountChanged: searchResults.isAddedContact = searchResults.isContactAdded()
 
             onContactClicked: {
-                root.store.changeAppSectionBySectionType(Constants.appSection.chat)
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Profile/views/ContactsView.qml:262]: Don't wanna bikeshed here, but wouldn't it still be preferred to have this API provided by the store to reduce dependencies in the view?

File: ui/app/AppLayouts/Wallet/WalletLayout.qml [hunk 2]
```diff
@@ -18,15 +18,15 @@ Item {
...
+        if(!hideSignPhraseModal && !RootStore.hideSignPhraseModal){
             signPhrasePopup.open();
         }
     }
 
     SignPhraseModal {
         id: signPhrasePopup
         onRemindLaterClicked: hideSignPhraseModal = true
-        onAcceptClicked: localAccountSensitiveSettings.hideSignPhraseModal = true
+        onAcceptClicked: RootStore.hideSignPhraseModal = true
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Wallet/WalletLayout.qml:29]: This `hideSignPhraseModal` mapped/aliased to `localAccountSensitiveSettings`?

We use setter functions to store this value in a config file (in `Status/qt/[account]`), so we need to make sure this still works when this value is changed.

File: ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml [hunk 3]
```diff
@@ -48,8 +49,8 @@ Item {
         font.pixelSize: 15
     }
     StyledText {
-        id: assetBalance
-        text: Utils.toLocaleString(balance, localAppSettings.locale) + " " + symbol.toUpperCase()
+        id: assetAddress
+        text: Utils.toLocaleString(currencyBalance.toFixed(2), locale) + " " + assetDelegate.currency.toUpperCase()
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml:53]: Are these three changes correct? We're changing what was the `assetFullTokenName` to `assetBalance` and what was `assetSymbol` to `assetFullTokenName` etc.

This looks like a change in logic. Is this intended?

File: ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml [hunk 4]
```diff
@@ -58,7 +59,8 @@ Item {
     StyledText {
         id: assetCurrencyBalance
         color: Style.current.secondaryText
-        text: Utils.toLocaleString(currencyBalance.toFixed(2), localAppSettings.locale) + " " + assetDelegate.currency.toUpperCase()
+        //TODO figure out what goes here
+        text: "%"
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml:63]: It's unclear to me why this change was made in the first place? Is there any reason we don't leave the value where they were?

File: ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml [hunk 5]
```diff
@@ -54,7 +54,7 @@ ModalPopup {
         anchors.horizontalCenter: parent.horizontalCenter
         width: 240
         dropdownWidth: parent.width - (Style.current.padding * 2)
-        dropdownAlignment: Select.MenuAlignment.Center
+        dropdownAlignment: StatusSelect.MenuAlignment.Center
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:57]: Good catch!

File: ui/app/AppLayouts/Wallet/popups/TokenSettingsModal.qml [hunk 6]
```diff
@@ -14,7 +14,9 @@ ModalPopup {
     id: popup
     //% "Manage Assets"
     title: qsTrId("manage-assets")
-    
+    signal addCustomTokenTriggered()
+    signal tokenDetailsTriggered(string address, string address, string address, string address)
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Wallet/popups/TokenSettingsModal.qml:18]: This looks like this wants to be changed.

When we listen to `onTokenDetailsTriggered` and access the `address` scoped argument, which one will we get?

**Review by PascalPrecht — CHANGES_REQUESTED** [review]
Massive.

Found a few things that might need a change.

File: ui/app/AppLayouts/Wallet/WalletLayout.qml (hunk 2 above)
Comment (jrainville) [reply ui/app/AppLayouts/Wallet/WalletLayout.qml:29]: Also, I know the way Sale created the settings was meant in a "globally available" way, so maybe passing through the Store for them is "overkill"?

File: ui/app/AppLayouts/Wallet/popups/AddCustomTokenModal.qml [hunk 7]
```diff
@@ -148,6 +148,13 @@ ModalPopup {
         height: addBtn.height
         visible: editable
 
+        MessageDialog {
+            id: changeError
+            title: "Changing settings failed"
```
Comment (jrainville) [inline ui/app/AppLayouts/Wallet/popups/AddCustomTokenModal.qml:153]:
```suggestion
            title: qsTr("Changing settings failed")
```

**Review by jrainville — COMMENTED** [review]
Nice job. I left 2 comments, but I also support Pascal in his comments

File: ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml (hunk 3 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml:53]: I just looked at the designs for wallet2 and thought updating, but reverted the change now.

File: ui/app/AppLayouts/Wallet/WalletLayout.qml (hunk 2 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Wallet/WalletLayout.qml:29]: @PascalPrecht if you check on wallet's RootStore there is code for this:
```
onHideSignPhraseModalChanged: {
    accountSensitiveSettings.hideSignPhraseModal = hideSignPhraseModal;
}
```
I could introduce another function for this if you want.

@jrainville yes correct, however this is way is not flexible if we want to have the UI running with a simulation backend, we cannot be introducing dependencies, that's why we use the stores to encapsulate things there and have a single place to update.

File: ui/app/AppLayouts/Profile/views/ContactsView.qml (hunk 1 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/views/ContactsView.qml:262]: `changeAppSectionBySectionType` is called by many places in the UI, hence making it a global function. If we move it to the store here, then each store should have it as well having us duplicating code. We could either do this, or keep it in the Global component which is it's single place to update if needed.

File: ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml (hunk 4 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml:63]: answered above

File: ui/app/AppLayouts/Wallet/popups/TokenSettingsModal.qml (hunk 6 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Wallet/popups/TokenSettingsModal.qml:18]: done :D

File: ui/app/AppLayouts/Wallet/popups/AddCustomTokenModal.qml (hunk 7 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Wallet/popups/AddCustomTokenModal.qml:153]: done, plus added in `AccountSettingsModal` as well.

**Review by jrainville — APPROVED** [review]
Nice work!

File: ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml (hunk 3 above)
Comment (PascalPrecht) [reply ui/app/AppLayouts/Wallet/controls/AssetDelegate.qml]: Thanks!

File: ui/app/AppLayouts/Wallet/WalletLayout.qml (hunk 2 above)
Comment (PascalPrecht) [reply ui/app/AppLayouts/Wallet/WalletLayout.qml:29]: Thanks for clarifying! @alexandraB99 if this keeps the same logic around, this is fine by me as is!

**Review by PascalPrecht — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/WalletLayout.qml (hunk 2 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Wallet/WalletLayout.qml]: @PascalPrecht I switched to use a setter function :)
